sdk/src/androidTest/java/com/mongodb/stitch/StitchClientServiceTest.kt

```diff
@@ -0,0 +1,141 @@
+package com.mongodb.stitch
+
+import android.content.Context
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import com.mongodb.stitch.android.StitchClient
+import com.mongodb.stitch.android.auth.apiKey.ApiKey
+import com.mongodb.stitch.android.auth.apiKey.ApiKeyProvider
+import com.mongodb.stitch.android.auth.emailpass.EmailPasswordAuthProvider
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Test public methods of StitchClient, running through authentication
+ * and pipeline flows. Full service test.
+ */
+@RunWith(AndroidJUnit4::class)
+class StitchClientServiceTest {
+    /** Base context from test runner */
+    private val instrumentationCtx: Context by lazy { InstrumentationRegistry.getContext() }
+    /** [StitchClient] for this test */
+    private val stitchClient: StitchClient by lazy {
+        StitchClient(
+                instrumentationCtx,
+                "{TEST-APP-NAME}",
```

Comment:
[testing] Let's make this settable when running the tests. Then we can update the evergreen file. I'm assuming tests will currently fail on evergreen right?